vaadin-grid-pro-flow-testbench/src/main/java/com/vaadin/flow/component/gridpro/testbench/GridProElement.java

```diff
@@ -98,6 +102,62 @@ public int getLastVisibleRowIndex() {
                 this)).intValue();
     }
 
+    /**
+     * Gets the header cell for the given visible column index.
+     *
+     * @param columnIndex
+     *            the index of the column
+     * @return a cell element for the header cell
+     */
+    public GridTHTDElement getHeaderCell(int columnIndex) {
+        return getVisibleColumns().get(columnIndex).getHeaderCell();
+    }
+
+    /**
+     * Finds the vaadin-grid-cell-content element for the given row and column
+     * in header.
+     *
+     * @param rowIndex
+     *            the index of the row in the header
+     * @param columnIndex
+     *            the index of the column in the header
+     * @return the vaadin-grid-cell-content element for the given row and column
+     *         in header.
+     */
+    public TestBenchElement getHeaderCellContent(int rowIndex,
+                                                 int columnIndex) {
+        WebElement thead = findInShadowRoot(By.id("header")).get(0);
+        List<WebElement> headerRows = thead.findElements(By.tagName("tr"));
+        List<WebElement> headerCells = headerRows.get(rowIndex)
+                .findElements(By.tagName("th"));
+        String slotName = headerCells.get(columnIndex)
+                .findElement(By.tagName("slot")).getAttribute("name");
+
+        return findElement(By.cssSelector(
+                "vaadin-grid-cell-content[slot='" + slotName + "']"));
+    }
+
+    /**
+     * Find all {@link WebElement}s using the given {@link By} selector.
+     *
+     * @param by
+     *            the selector used to find elements
+     * @return a list of found elements
+     */
+    public List<WebElement> findInShadowRoot(By by) {
```

Comment:
Some issues with this method:
- This seems like a generic feature for `TestBenchElement`, so if we want to provide it, it should not be specific to `GridPro`
- It returns a list of `WebElements` which is weird for testbench API, since everything else uses the `TestBenchElement` wrapper
- The testbench query method `$` already searches for elements in the shadow root, so I don't see a reason to add a specific method that searches only from the shadow root

So I would just use `$` and remove this method.

Reply:
Valid points, simplified now.